components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/config/OutboundAdapterConfigurationProvider.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.identity.event.websubhub.publisher.config;
+
+import edu.umd.cs.findbugs.annotations.SuppressWarnings;
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.identity.event.common.publisher.exception.AdapterConfigurationException;
+import org.wso2.identity.event.common.publisher.exception.AdapterRuntimeException;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.channels.Channels;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Optional;
+import java.util.Properties;
+
+import static java.util.Objects.isNull;
+import static java.util.Optional.empty;
+import static java.util.Optional.ofNullable;
+import static org.wso2.identity.event.websubhub.publisher.constant.WebSubHubAdapterConstants.CONFIG_FILE_NAME;
+
+/**
+ * Class to build the output adapter configurations.
+ */
+public class OutboundAdapterConfigurationProvider {
+
+    private static final OutboundAdapterConfigurationProvider instance;
+
+    static {
+        try {
+            instance = new OutboundAdapterConfigurationProvider();
+        } catch (AdapterConfigurationException e) {
+            throw new AdapterRuntimeException("Error initializing outbound adapter configuration provider.", e);
+        }
+    }
```

Comment:
Rather than having a static block why not we have a singleton and invoke it from where initialization needs to happen (maybe as the bundle start)

Reply:
Fixed with https://github.com/wso2-extensions/identity-event-publishers/pull/1/commits/7e62e79fe00e7222402c1e7508d0ce4d065b24d0